android/java/org/chromium/chrome/browser/ntp/sponsored/NTPUtil.java

```diff
@@ -0,0 +1,269 @@
+                parentLayout.addView(mainLayout);
+
+                parentLayout.setOrientation(LinearLayout.HORIZONTAL);
+
+                LinearLayout.LayoutParams mainLayoutLayoutParams = new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.MATCH_PARENT);
+                mainLayoutLayoutParams.weight = 0.6f;
+                mainLayout.setLayoutParams(mainLayoutLayoutParams);
+
+                LinearLayout.LayoutParams imageCreditLayoutParams = new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.MATCH_PARENT);
+                imageCreditLayoutParams.weight = 0.4f;
+                imageCreditLayout.setLayoutParams(imageCreditLayoutParams);
+
+                layoutParams.setMargins(dpToPx(context,32), 0, 0, 0);
+                layoutParams.gravity = Gravity.BOTTOM | Gravity.START;
+                sponsoredLogo.setLayoutParams(layoutParams);
+            }
+        } else {
+            // In portrait
+            parentLayout.removeView(mainLayout);
+            parentLayout.removeView(imageCreditLayout);
+
+            parentLayout.addView(mainLayout);
+            parentLayout.addView(imageCreditLayout);
+
+            parentLayout.setOrientation(LinearLayout.VERTICAL);
+
+            LinearLayout.LayoutParams mainLayoutLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,0);
+            mainLayoutLayoutParams.weight = 1f;
+            mainLayout.setLayoutParams(mainLayoutLayoutParams);
+
+            LinearLayout.LayoutParams imageCreditLayoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+            imageCreditLayout.setLayoutParams(imageCreditLayoutParams);
+
+            layoutParams.gravity = Gravity.BOTTOM | Gravity.CENTER_HORIZONTAL;
+            sponsoredLogo.setLayoutParams(layoutParams);
+        }
+    }
+
+    public static int checkForNonDistruptiveBanner(NTPImage ntpImage, SponsoredTab sponsoredTab) {
+        BraveRewardsNativeWorker mBraveRewardsNativeWorker = BraveRewardsNativeWorker.getInstance();
+        if (sponsoredTab.shouldShowBanner()) {
+            if (BraveRewardsPanelPopup.isBraveRewardsEnabled()) {
+                if (BraveAdsNativeHelper.nativeIsBraveAdsEnabled(Profile.getLastUsedProfile())) {
+                    if (ntpImage instanceof SponsoredImage) {
+                        return SponsoredImageUtil.BR_ON_ADS_ON;
+                    }
+                } else if(BraveAdsNativeHelper.nativeIsLocaleValid(Profile.getLastUsedProfile())) {
+                    if (ntpImage instanceof SponsoredImage) {
+                        return SponsoredImageUtil.BR_ON_ADS_OFF ;
+                    } else {
+                        return SponsoredImageUtil.BR_ON_ADS_OFF_BG_IMAGE;
+                    }
+                }
+            } else {
+                if (ntpImage instanceof SponsoredImage && !mBraveRewardsNativeWorker.IsCreateWalletInProcess()) {
```

Comment:
why do we care if the wallet creation is in process?

Reply:
from bottom banner, user can enable rewards and according to current implementation, it takes some time to create wallet and enable rewards. and if you open another tab right after enabling rewards from the bottom banner, it shows the banner again as the state is disabled. so this check is to see whether the wallet creation is in progress, if it is then we don't show the banner again